packages/packages/v1-adapters/src/listeners/listeners.ts

```diff
@@ -0,0 +1,79 @@
+import { normalizeEvent } from './utils';
+import { CommandEventDescriptor, EventDescriptor, ListenerCallback, WindowEventDescriptor } from './types';
+
+const callbacksByEvent = new Map<EventDescriptor['name'], ListenerCallback[]>();
+let abortController = new AbortController();
+
+export function listenTo(
+	eventDescriptors : EventDescriptor | EventDescriptor[],
+	callback : ListenerCallback
+) {
+	if ( ! Array.isArray( eventDescriptors ) ) {
+		eventDescriptors = [ eventDescriptors ];
+	}
+
+	eventDescriptors.forEach( ( event ) => {
+		const { type, name } = event;
+
+		switch ( type ) {
+			case 'command':
+				registerCommandListener( name, event.state, callback );
```

Comment:
Consider creating one interface for both events (and the state might be undefined).

Reply:
feels kinda odd 🤷